### ISSUE TITLE
java.lang.StackOverflowError for array[8192]

### ISSUE THREAD

**Issue by invd** [ultimate-pa/ultimate#366]
Example program with straightforward out of bounds memory access:
```
//#Unsafe
typedef unsigned int uint8_t;
static uint8_t _oledbuffer[8192];

int main() {
  // fault
  _oledbuffer[-1];
  return 0;
}
```

After less than 2 minutes of runtime, Ultimate runs into smtinterpol errors:
`./Ultimate -tc ~/ultimate/releaseScripts/default/UAutomizer-linux/config/AutomizerMemDerefMemtrack.xml -s ~/ultimate/trunk/examples/settings/default/automizer/svcomp-DerefFreeMemtrack-32bit-Automizer_Default.epf -i ~/source/code/experiments/backtrace_array_verbosity.c`
```
[2018-07-06 16:43:14,134 FATAL L292        ToolchainWalker]: The Plugin de.uni_freiburg.informatik.ultimate.plugins.generator.traceabstraction has thrown an exception:                                            
java.lang.StackOverflowError                 
        at de.uni_freiburg.informatik.ultimate.smtinterpol.theory.cclosure.CCAppTerm.toSMTTerm(CCAppTerm.java:204)  
[this error line repeats itself]
```
Variants of the example program with `_oledbuffer[1024]` and `_oledbuffer[4096]` can be analyzed **without** this error.

The affected Ultimate version is the recent 8de8874bb85c5cb95aa1930e149dec9dbc679170 and the Java memory settings are default: ` -Xmx12G -Xms512M`. The target VM has ~19GB of RAM.

**Comment by invd** [comment]
I can confirm that this error can be triggered by `_oledbuffer[6000];` as well.

**Comment by danieldietsch** [comment]
@alexandernutz can the new initialization help for this example?

**Comment by invd** [comment]
I can confirm that Ultimate version 472692297f66b14197f63de46ccd9873a7dc260c from yesterday is still affected.

**Comment by jhoenicke** [comment]
I can look into making these conversions non-recursive.  Looks like it happens in proof production.

The array initialization would help, but it is not ready yet, and we have no idea how to interpolate some special cases for the new proof rules required for constant arrays.

**Comment by jhoenicke** [comment]
@invd I can't reproduce it (for me z3 runs out of memory).  Can you generate the smt scripts? I think the option is `--traceabstraction.dump.smt.script.to.file true`

**Comment by invd** [comment]
Here is the corresponding smt2 file (renamed as .smt2.txt) obtained via `--t
raceabstraction.dump.smt.script.to.file true `: [Script_backtrace_array_verbosity_6000.smt2.txt](https://github.com/ultimate-pa/ultimate/files/2245539/Script_backtrace_array_verbosity_6000.c_Iterationbacktrace_array_verbosity_6000.c_AllErrorsAtOnce_1.smt2.txt)

**Comment by jhoenicke** [comment]
Stack overflow fixed by ultimate-pa/smtinterpol@571181f.

**Comment by danieldietsch** [comment]
Merged into Ultimate with 474129a

@invd: Do you want to recheck?

**Comment by danieldietsch** [comment]
Ok, this sounds like a different issue then. New modelling for array initialization (@alexandernutz is working on them) may help. 
But the stack overflow is gone, so I am closing this ticket.
